### PR TITLE
fix issues associated with apply_theme() function

### PR DESCRIPTION
**Pull request by SHAESEN2**
**What changes are proposed in this pull request?**
Ensure that only the strata present in the theme are displayed.
Other strata are greyed out and legend title is aligned with theme.

**Which files are involved in this pull request and what changes were made?**

**Did you include unit tests for the proposed change/bug fix (https://testthat.r-lib.org/)?**
Pending

**If there is an GitHub issue associated with this pull request, please provide link.**
#388 
--------------------------------------------------------------------------------

Checklist for PR reviewer

- [x] PR branch has pulled the most recent updates from main branch. Ensure the pull request branch and your local version match and both have the latest updates from the main branch.
- [x] If a new function was added, function should be included in `_pkgdown.yml`
- [x] If a bug was fixed, a unit test was added for the bug check
- [x] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [x] Code coverage is suitable for any new functions/features. Review coverage with `withr::with_envvar(new = c("NOT_CRAN" = "true"), covr::report())`. Before you run, begin a fresh R session without any packages loaded. 
- [x] R CMD Check runs without errors, warnings, and notes
- [x] `usethis::use_spell_check()` runs with no spelling errors in documentation
- [x] Has `NEWS.md` been updated with the changes from this pull request under the heading indicating the latest version. If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [x] Has the version number been incremented using `usethis::use_version(which = "dev")` 
- [x] Has the README been knitted before merging to ensure each PR has the most up-to-date readme?
- [x] Approve Pull Request
- [x] Merge the PR. Please use "Squash and merge".

### CODE REVIEW

**Comment by ddsjoberg** [comment]
@SHAESEN2 i think i can review later tonight. in the meantime, the R CMD Checks are not passing. Do you want to take a look before it's reviewed?

File: R/apply_theme.R [hunk 1]
```diff
@@ -443,9 +457,15 @@ apply_theme <- function(gg, visR_theme_dict = NULL) {
   # Reset background
   gg <- gg + ggplot2::theme_minimal()
 
+  if (!skipcolor) {
+    gg <- gg +
+      ggplot2::scale_colour_manual(labels = names(cols),
+                                   values = cols,
+                                   aesthetics = c("colour", "fill"), na.value = "grey50") +
+      guides(color=guide_legend(ttl))
```
Comment (bailliem) [inline R/apply_theme.R:465]: add ggplot2 to guides on line 465 

`ggplot2::guides(color=guide_legend(ttl))`

**Comment by bailliem** [comment]
There is an error being thrown in the Styling_KM_Plots.Rmd vignette running the chunk on line 135

```{r visr-apply_theme-empty, warning=FALSE}
lung_suvival_object %>%
  visR::visr() %>%
  apply_theme()
```

Error in apply_theme(.) : object 'skipcolor' not found
2. apply_theme(.)
1. lung_suvival_object %>% visR::visr() %>% apply_theme()

**Review by bailliem — CHANGES_REQUESTED** [review]

**Comment by SHAESEN2** [comment]
> > There is an error being thrown in the Styling_KM_Plots.Rmd vignette running the chunk on line 135
> > ```
> > lung_suvival_object %>%
> >   visR::visr() %>%
> >   apply_theme()
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Error in apply_theme(.) : object 'skipcolor' not found 2. apply_theme(.)
> > 
> > 1. lung_suvival_object %>% visR::visr() %>% apply_theme()
> 
> Ok. If I understand the example its apply an empty theme expecting to revert to the default theme. I'm not sure the example makes a lot of sense. Why would you apply_theme() that is null. Maybe the "fix" here is a warning if no user supplied theme is provided as an arguement.

It is to be able to apply sensible defaults as per good visualization practices. However I feel that we should implement them directly rather than making this complex function.

**Comment by SHAESEN2** [comment]
I think we should add experimental to the function + documentation is still pending

**Review by bailliem — APPROVED** [review]

**Comment by bailliem** [comment]
> I think we should add experimental to the function + documentation is still pending

Agree. This seems like a nice to have for now.

**Comment by timtreis** [comment]
I added a test that covers the behaviour when not enough colours are given and "grey50" is substituted. I'll let Codecov tell me if there's more untested behaviour, but other than that it looks good!

**Comment by timtreis** [comment]
image

I think this warning can never be triggered because the required situation (more colours than in the visR palette) is caught here and the `else if (!skipcolordef)` is never entered.
image

**Comment by SHAESEN2** [comment]
> I added a test that covers the behaviour when not enough colours are given and "grey50" is substituted. I'll let Codecov tell me if there's more untested behaviour, but other than that it looks good!
> 
> Edit: Now testing for two more use-cases you covered:
> 
> a) Use visR palette when theme has no info on strata used in plot (-> had to set `skipcolordef` to `FALSE`) b) Use ggplot palette when strata has more levels than the visR palette elements

Thanks for taking the time!!! I'll go through the testing and let you review it once again.

**Review by timtreis — APPROVED** [review]

File: R/apply_theme.R [hunk 2]
```diff
@@ -1,178 +1,4 @@
...
-
-  }
-
-  base::class(theme) <- c( "visR_theme", class(theme))
-
-  return(theme)
-
-}
-
+#' `r lifecycle::badge("experimental")`
```
Comment (bailliem) [inline R/apply_theme.R:1]: Move to under description to help with documentation rendering.

File: R/define_theme.R [hunk 3]
```diff
@@ -0,0 +1,175 @@
+#' `r lifecycle::badge("experimental")`
```
Comment (bailliem) [inline R/define_theme.R:1]: move under description

**Review by bailliem — APPROVED** [review]
Looks good - I'll push a couple of minor changes to the document strings then merge the PR

**Pull request merged**